middleware/data_source_queries.py

```diff
@@ -163,8 +163,39 @@ def data_sources_results(conn):
     return results
 
 
-def data_sources_query(conn={}, test_query_results=[]):
-    results = data_sources_results(conn, "", "") if conn else test_query_results
+def needs_identification_results(conn):
```

Comment:
"needs_identification_results" is a tad confusing as a function name. From the name, I would expect it to return a boolean (i.e., whether it needs identification results or not), although the body of the function makes it clear that is not the case. Recommend renaming this (perhaps to "get_needs_identification_results"? and adding type hinting (especially for the return value) for additional clarity.